tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1530,6 +1530,36 @@ Here are some examples of expressions::
 Identifiers, attributes, and subscripts are valid assignment targets. Other expressions cannot be on the left side of
 an assignment.
 
+self
+^^^^
+
+``self`` can be used to refer to the current instance and is often equivalent to directly referring to symbols available in
+the current script. However, ``self`` also allows you to access properties, methods, and other names that are defined dynamically
+(i.e. are expected to exist in subtypes of the current class, or are provided using :ref:`_set() <class_Object_private_method__set>`
+and/or :ref:`_get() <class_Object_private_method__get>`).
+
+::
+
+    extends Node
+
+    func _ready():
+        # Compile time error, as `my_var` is not defined in the current class or its ancestors.
+        print(my_var)
+        # Checked at runtime, thus may work for dynamic properties or descendant classes.
+        print(self.my_var)
+
+        # Compile time error, as `my_func()` is not defined in the current class or its ancestors.
+        my_func()
+        # Checked at runtime, thus may work for descendant classes.
+        self.my_func()
+
+.. warning::
+
+    Beware that accessing members of child classes in the base class is often considered a bad practice because this
+    makes the relationships between parts of your game's code harder to reason about. It is easy to forget to define
+    a variable that the base class expected or forget about an invariant that the base class assumed to be true when
+    implementing a function needed by the base class.
```

Comment:
The first sentence looks good, but I doubt the second one is a good reasoning/illustration. I would suggest removing it or replacing it to take the emphasis off "easy to forget".

I don't think it's considered bad practice because it's easy to forget to add a required member when inheriting. In my opinion, the main problem is that the class is accessing non-existent properties/methods that are outside its area of ​​responsibility. Subclasses are in a sense undefined until the base class is fully defined, so it shouldn't know about their existence.

If you want to use abstract methods (which currently don't exist in GDScript), it's better to define a "prototype" with an empty body or an assertion.

Reply:
Makes perfect sense, i included your reasoning into the `:: warning` section!